dev/com.ibm.ws.cdi.ejb_fat/fat/src/com/ibm/ws/cdi/ejb/tests/EjbTimerTest.java

```diff
@@ -0,0 +1,134 @@
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.EjbSessionBean2.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.view.EjbSessionBeanLocal.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.view.SessionBeanLocal.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.view.EjbSessionBean2Local.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.ApplicationScopedCounter.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.SessionBean.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.TestEjbNoTimerServlet.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.RequestScopedBean.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.TestEjbTimerServlet.class)
+                                        .addClass(com.ibm.ws.cdi.ejb.apps.timer.EjbSessionBean.class)
+                                        .add(new FileAsset(new File("test-applications/" + EJB_TIMER_APP_NAME + ".war/resources/META-INF/permissions.xml")),
+                                             "/META-INF/permissions.xml")
+                                        .add(new FileAsset(new File("test-applications/" + EJB_TIMER_APP_NAME + ".war/resources/WEB-INF/beans.xml")), "/WEB-INF/beans.xml");
+
+        ShrinkHelper.exportDropinAppToServer(server, ejbTimer, DeployOptions.SERVER_ONLY);
+        server.startServer();
+    }
+
+    /**
+     * Verifies that a Session Scoped counter works correctly when incremented via either a
+     * EJB Timer (i.e asynchronously)
+     *
+     * @throws Exception
+     *             if counter is wrong, or if an unexpected error occurs
+     */
+    @Test
+    public void testCDIScopeViaEJBTimer() throws Exception {
+        WebBrowser wb = WebBrowserFactory.getInstance().createWebBrowser();
+        //the count values returned are from BEFORE the increment occurs
+        //request count should always be 0 since it should be a new request each time
+
+        //first couple of times is synchronous (no timer or task)
+        verifyResponse(wb, server, "/ejbTimer/NoTimer", "session = 0 request = 0");
+        verifyResponse(wb, server, "/ejbTimer/NoTimer", "session = 1 request = 0");
+
+        //the next couple start a timer which will increment asynchronously after 1 second
+        //only one timer can be active at a time so subsequent calls will block until the previous timers have finished
+        verifyResponse(wb, server, "/ejbTimer/Timer", "session = 2 request = 0");
+        verifyResponse(wb, server, "/ejbTimer/Timer", "session = 3 request = 0");
+        verifyResponse(wb, server, "/ejbTimer/NoTimer", "session = 4 request = 0");
```

Comment:
Good news: it was reasonably easy to find the classes which implement this test.

Bad news: the call to `/ejbTimer/Timer` appears to increment the session scoped counter after setting up a timer, rather than when that timer executes, contrary to the javadoc on this test method. (See `EjbSessionBean#incCountersViaTimer`)

Reply:
I don't think that aspect of the test has changed ... so an issue should be opened to fix it but shouldn't stop this PR.

Reply:
https://github.com/OpenLiberty/open-liberty/issues/15789